RuntimeUnityEditor/Windows/ObjectTree/ObjectTreeViewer.cs

```diff
@@ -102,27 +103,34 @@ private void DisplayObjectTreeHelper(GameObject go, int indent, ref int currentC
         {
             currentCount++;
 
-            var needsHeightMeasure = _singleObjectTreeItemHeight == 0;
+            var needsHeightMeasure = _singleObjectTreeItemHeight == 0f;
 
-            var isVisible = currentCount * _singleObjectTreeItemHeight >= _treeScrollPosition.y &&
-                            (currentCount - 1) * _singleObjectTreeItemHeight <= _treeScrollPosition.y + _objectTreeHeight;
+            var isVisible = currentCount * _singleObjectTreeItemHeight + _singleObjectTreeItemMargin >= _treeScrollPosition.y &&
+                            (currentCount - 1) * _singleObjectTreeItemHeight + _singleObjectTreeItemMargin <= _treeScrollPosition.y + _objectTreeHeight;
 
-            if (isVisible || needsHeightMeasure || _scrollTreeToSelected)
+            if (SelectedTransform == go.transform && _scrollTreeToSelected)
```

Comment:
No need to display every element to find scrollTarget. Now that we have deterministic calculation, can predict exact scrollTarget